### ISSUE TITLE
Inaccurate music in-game?

### ISSUE THREAD

**Issue by amysgames** [chrismaltby/gb-studio#83]
Hello! So I've been experimenting with this neat tool, and I've noticed a few issues with in-game music. Namely, it doesn't seem like it's playing notes in the correct octave, and it's not detecting tempo changes at all. This only applies when I run the game; the editor itself plays the music no problem. I'm not sure if it's compatibility issues with MilkyTracker, GB hardware constraints I'm not aware of, or something else?

**Comment by WAUthethird** [comment]
Open the .mod file in OpenMPT and see if the issues arise or not.

**Comment by amysgames** [comment]
It sounds the same in OpenMPT as it does in MilkyTracker. I tried saving it in OpenMPT and that didn't help.

I'm also fairly new to tracking, myself, so I'm not sure if maybe I did something wrong?

**Comment by WAUthethird** [comment]
Hmm, I do know that the Hz of the song is changed when compiling the song, so it's normal if it sounds a bit higher or sped-up.
Are the differences very drastic?

**Comment by amysgames** [comment]
It's definitely noticeable. It's playing it a full octave lower than what plays in MilkyTracker, and it disregards my tempo settings entirely (I tried setting it to something different with no change).

The instrument choices are drastically different as well, though I'd understand if there's some hardware limitations there.

**Comment by JasonB32** [comment]
I'm having the exact same issue and I had no idea what was going on, I use OpenMPT as well, and yes Milkytracker plays it fine too, but GB Studio through GBT Player is playing it slower and lower :(

**Comment by RichardULZ** [comment]
I'll make a tutorial on openMTP with Mod2GBT eventually, but right now, here's all the documentation.
Limits of mod2gbt https://github.com/AntonioND/gbt-player/blob/master/mod2gbt/mod_instructions.txt
Always start from a base file, instruments must have same layout. Download effects_test.mod https://github.com/AntonioND/gbt-player/tree/master/mod2gbt
A complete guide from zalo's ZGB which uses the same GBT player.

**Comment by JasonB32** [comment]
Oh if you could please do a tutorial that would be great!!!

**Comment by amysgames** [comment]
Oh, so it _is_ a compatibility thing, then. Alright, good to know.

With those restrictions in mind, I've managed to put together something that sounds a lot closer to what I want, but I guess there's not much I can do about the incorrect tempo (speed 5 is too fast and speed 6 is too slow). Oh well.

Thank you!

**Comment by RichardULZ** [comment]
Try to space your song out differently! 
If your song is lacking depth, try out stereo panning, and fade the volume of any instruments over time with volume commands to get them more natural.

Tracker ticks often run at 50hz, but the Gameboy is 60, so GBT does a funky conversion from 50 to 60 for player consistency, Haven't tried to disable that yet, but it's possible...

**Comment by Digital-Down** [comment]
Similar problem here, sounds are much higher pitched in GBSTUDIO as opposed to the tracker and notes are sometimes more consistent than others. Sometimes completely unrecognizable. I’ll read up on the documentation posted in a prior comment.

**Comment by timsamoff** [comment]
Just chiming in (no pun intended) to ask if there's a more GUI method of creating the proper Mod files... I have downloaded several Mods -- and created a couple of my own (presumably via incompatible means) -- and none of them play correctly in GB Studio. For instance, test the following Mod (not mine), made with [Music Box](http://gbdev.gg8.se/files/musictools/Music%20Box/), which is software that should output GB compatible Mod files:

[catg2.mod.zip](https://github.com/chrismaltby/gb-studio/files/3163989/catg2.mod.zip)

I don't mind using Tracker software, but nothing seems to come out right.

**Comment by RichardULZ** [comment]
@timsamoff you must open a template or example file designed for gb studio or mod2gbt, they are set up in a very specific way with instuments that must not be modified.
I highly recommend openMTP as its a more windows like GUI than most trackers. It is still like learning a tracker.

Read all the document links i posted, the zalo one is a good start, and for audio issues, the mod_instructions.txt is your trouble shooter guide.

**Comment by timsamoff** [comment]
@RichardULZ: Thanks, but like I said, I'm looking for something comparable for Mac.

**Comment by WAUthethird** [comment]
@timsamoff: Try [MilkyTracker](https://milkytracker.titandemo.org/downloads/).

**Comment by timsamoff** [comment]
@WAUthethird: I did, but maybe I just messed up the template instruments somehow. I'll give it another go if it's the only option.

**Comment by WAUthethird** [comment]
@timsamoff If nothing else, there's always WINE.

**Comment by timsamoff** [comment]
@WAUthethird: Yes. I have a Windows machine as well. I was just trying to keep everything going on one. ;)

**Comment by WAUthethird** [comment]
@timsamoff I see. Hope MilkyTracker works out for you, then.

**Comment by timsamoff** [comment]
@RichardULZ and @WAUthethird: I have composed some music that is working perfectly now, EXCEPT that none of the tempo data is saved in the file. Is this possible? Or am I stuck at 125 BPM?

**Comment by timsamoff** [comment]
> @RichardULZ and @WAUthethird: I have composed some music that is working perfectly now, EXCEPT that none of the tempo data is saved in the file. Is this possible? Or am I stuck at 125 BPM?

Sorry! Just figured out the Fxx commands!
